registration/include/pcl/registration/impl/sample_consensus_prerejective.hpp

```diff
@@ -234,10 +233,17 @@ SampleConsensusPrerejective<PointSource, PointTarget, FeatureT>::computeTransfor
     }
   }
 
-  // Feature correspondence cache
+  // Precompute feature correspondence cache
   std::vector<std::vector<int> > similar_features (input_->size ());
+  std::vector<float> nn_distances(k_correspondences_);
+  #pragma omp parallel for num_threads(num_threads_) firstprivate(nn_distances)
+  for(int i = 0; i < static_cast<int>(input_->size()); ++i) {
+    nn_distances.clear();
+    feature_tree_->nearestKSearch(*input_features_, i, k_correspondences_, similar_features[i], nn_distances);
+  }
 
   // Start
+  #pragma omp parallel for num_threads(num_threads_) shared(similar_features) private(inliers, error)
```

Comment:
You can do the same (pulling out of loop and marking as firstprivate) for the following: `sample_indices` and `corresponding_indices`

Reply:
Done